src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -158,20 +158,20 @@ private void internalAddrow(JDBCType jdbcType,
 	        int nValueLen;
 	        switch (jdbcType) {
 	            case BIGINT:
-	                rowValues[pair.getKey()] = (val instanceof Number ? ((Number)val).longValue() : Long.parseLong(val.toString()));
+	                rowValues[pair.getKey()] = (val instanceof Number ? (Long)val : Long.parseLong(val.toString()));
```

Comment:
If `val` is an `Integer` this will throw a `ClassCastException`. Same for most of the `instanceof` checks below